sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -738,13 +801,13 @@ def _add_argparse_args(cls, parser):
               '"<ENV_VAL>"} }. All fields in the json are optional except '
               'command.'))
     parser.add_argument(
-        '--sdk-worker-parallelism', default=None,
+        '--sdk_worker_parallelism', default=None,
```

Comment:
Would not this change be backward incompatible? Does argparse do an auto conversion?

Reply:
Argparse will use the same key for storing flags with dashes and flags with underscores, but argparse will not consider them the same flags.  See also my comment above: https://github.com/apache/beam/pull/8225#discussion_r274140134

Reply:
So if we declare a flag with dashes, argparse will ignore a flag with underscores passed on the command line.

Reply:
This is a bit confusing.

Would a user's pipeline break if they have been using the 'sdk-worker-parallelism' version in command line or by directly passing to pipeline options in kwargs?

Reply:
Yes, this would have been a backwards-incompatible change. Luckily, looks like sdk-worker-parallelism didn't make it into 2.12.0: https://github.com/apache/beam/blob/release-2.12.0/sdks/python/apache_beam/options/pipeline_options.py#L719.

Reply:
The two options with dashes were added in https://github.com/apache/beam/pull/8082

Reply:
This may help clarify argparse behavior:
```
In [1]: import argparse

In [2]: parser=argparse.ArgumentParser()

In [3]: parser.add_argument('--flag-with-dash')
Out[3]: _StoreAction(option_strings=['--flag-with-dash'], dest='flag_with_dash', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)                                          

In [5]: parser.parse_args(['--flag-with-dash=abc'])
Out[5]: Namespace(flag_with_dash='abc')

In [6]: parser.parse_args(['--flag_with_dash=I am using underscores instead of dashes in flag name!'])
usage: ipython [-h] [--flag-with-dash FLAG_WITH_DASH]
ipython: error: unrecognized arguments: --flag_with_dash=I am using underscores instead of dashes in flag name!
An exception has occurred, use %tb to see the full traceback.

In [7]: parser.add_argument('--flag_with_dash')
Out[7]: _StoreAction(option_strings=['--flag_with_dash'], dest='flag_with_dash', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)                                          

In [8]: parser.parse_args(['--flag_with_dash=Now I can use both dashes and underscores'])
Out[8]: Namespace(flag_with_dash='Now I can use both dashes and underscores')

In [9]: parser.parse_args(['--flag-with-dash=Same store is used for flags with dashes and underscores'])
Out[9]: Namespace(flag_with_dash='Same store is used for flags with dashes and underscores')
```

Reply:
Awesome, sounds like we are safe.